### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by Temitopefa**
implement profile page design

### CODE REVIEW

File: src/pages/UserProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { APICard, UserProSideBar , TabPanel} from '../components'
+import { useContextProvider } from "../contexts/ContextProvider";
+
+
+import { makeStyles } from '@mui/styles';
+import { Avatar, Stack, Typography, Tabs, Tab} from '@mui/material'
+import { APIS } from "../testdata";
+
+
+const UserProfile = () => {
```
Comment (pablo-clueless) [inline src/pages/UserProfile.tsx]: Add a type `React.FC` to this component.

File: src/pages/UserProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                <Stack>
+                    <Tabs  value={tab} onChange={handleTabSwitch}>
+                    <Tab  className={classes.tabs} label= {`Published APIs (${list})`} />
+                    <Tab className={classes.tabs} label='Subscribed APIs'/>
+                    <Tab className={classes.tabs} label='Followed By(0)' />
+                    <Tab className={classes.tabs} label='Following(0)'/>
+                    </Tabs>
+                    <Stack mt={2}>
+                    <TabPanel value={tab} index={0}>
+                    <div style={{width:"100%",height:"60vh",display:"flex",flexWrap:"wrap",alignItems:"center",justifyContent:"center",gap:"1.5rem",overflowY:"scroll",padding:"0.5rem 0"}}>
```
Comment (taiwoAkindele) [inline src/pages/UserProfile.tsx]: We don't need `height` here. The cards will only occupy a certain part of the page by setting the height.

File: src/components/UserProSideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+const UserProSideBar:React.FC = ( ) => {
+    const classes = useStyles();
+
+
+    return(
+        <>
+            <div className={classes.sidebar}>
+                <Stack direction='column'>
```
Comment (pablo-clueless) [inline src/components/UserProSideBar.tsx]: This approach adds to many unnecessary div to the DOM, the sidebar style is already a column, adding `Stack` with column just adds too many load.

File: src/components/UserProSideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    <Stack style={{ alignItems: 'center'}}>
+                        <Avatar alt='Dummy-image'  sx={{ width: 150, height: 150 }} />
```
Comment (pablo-clueless) [inline src/components/UserProSideBar.tsx]: Create a dummy user and use the data so you can clearly see what you're working with.

**Review by taiwoAkindele — APPROVED** [review]

**Comment by taiwoAkindele** [comment]
This branch has conflicts on it. Please resolve the conflicts

**Review by pablo-clueless — APPROVED** [review]

**Pull request merged**